src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala

```diff
@@ -151,19 +151,54 @@ class SyncControllerSpec extends FlatSpec with Matchers with BeforeAndAfter with
 
     sendBlockHeaders(
       defaultTargetBlockHeader.number,
-      Seq(defaultTargetBlockHeader.copy(number = defaultExpectedTargetBlock - 1)),
+      Seq(defaultTargetBlockHeader.copy(number = defaultExpectedTargetBlock)),
       peer1,
       defaultExpectedTargetBlock - bestBlockNumber)
 
     persistState()
 
     val syncState = storagesInstance.storages.fastSyncStateStorage.getSyncState().get
 
-    syncState.bestBlockHeaderNumber shouldBe (bestBlockNumber - syncConfig.fastSyncBlockValidationN - 1)
-    syncState.nextBlockToFullyValidate shouldBe (bestBlockNumber - syncConfig.fastSyncBlockValidationN)
+    syncState.bestBlockHeaderNumber shouldBe (bestBlockNumber - syncConfig.fastSyncBlockValidationN)
+    syncState.nextBlockToFullyValidate shouldBe (bestBlockNumber - syncConfig.fastSyncBlockValidationN + 1)
     syncState.blockBodiesQueue.isEmpty shouldBe true
     syncState.receiptsQueue.isEmpty shouldBe true
+  }
+
+  it should "not change best block after receiving faraway block" in new TestSetup(validators = new Mocks.MockValidatorsAlwaysSucceed {
+    override val blockHeaderValidator: BlockHeaderValidator = { (blockHeader, getBlockHeaderByHash) => Left(HeaderParentNotFoundError) }
+  }) {
+
+    val targetNumber = 200000
+    val bestNumber = targetNumber - 10
+    val startState = defaultState.copy(
+      targetBlock = baseBlockHeader.copy(number = targetNumber ),
+      bestBlockHeaderNumber = bestNumber,
+      safeDownloadTarget = targetNumber + 10
+    )
+
+    startWithState(startState)
+    Thread.sleep(1.seconds.toMillis)
```

Comment:
Why do we sleep 1s here and 2s below?

Reply:
Actually when I used 1s,  fastsync was not quick enough (at least one time) to ask for the headers before I sent this future block. (it was probably one-time flicker but I tend to be pretty conservative in case of this `sleeps`, especially as they tend to fail especially on circleci)